File: fintrack-backend/src/main/java/com/fintrack/fintrackbackend/dto/GoalRequest.java
package com.fintrack.fintrackbackend.dto;

import jakarta.validation.constraints.DecimalMin;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

import java.math.BigDecimal;

public class GoalRequest {

    @NotBlank(message = "El nombre es obligatorio")
    private String name;

    @NotNull(message = "El objetivo es obligatorio")
    @DecimalMin(value = "0.01", message = "El objetivo debe ser mayor a 0")
    private BigDecimal target;

    private String description;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public BigDecimal getTarget() {
        return target;
    }

    public void setTarget(BigDecimal target) {
        this.target = target;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }
}

File: frontend/app/src/main/java/pe/fintrack/mobile/ui/theme/data/model/GoalRequest.kt
package pe.fintrack.mobile.ui.theme.data.model

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass
import java.math.BigDecimal

@JsonClass(generateAdapter = true)
data class GoalRequest(
    @Json(name = "name") val name: String,
    @Json(name = "target") val target: BigDecimal,
    @Json(name = "description") val description: String?
)
